cmd/influxd/run.go

```diff
@@ -45,8 +45,33 @@ type Node struct {
 	raftLog  *raft.Log
 
 	adminServer     *admin.Server
-	clusterListener net.Listener // The cluster TCP listener
-	apiListener     net.Listener // The API TCP listener
+	clusterListener net.Listener      // The cluster TCP listener
+	apiListener     net.Listener      // The API TCP listener
+	GraphiteServers []graphite.Server // The Graphite Servers
+	OpenTSDBServer  *opentsdb.Server  // The OpenTSDB Server
+}
+
+func (s *Node) ClusterAddr() net.Addr {
+	return s.clusterListener.Addr()
+}
+
+func (s *Node) ClusterURL() *url.URL {
+	h, p, e := net.SplitHostPort(s.ClusterAddr().String())
+	if e != nil {
+		panic(e)
+	}
+	if h == "::" || h == "" {
+		// Detect hostname (or set to localhost).
+		if h, _ = os.Hostname(); h == "" {
+			h = "localhost"
+		}
+	}
```

Comment:
on OSX, if i run `influxd -hostname localhost`, `h` will initially be `"::"` and then will become the result of `os.Hostname()` when what i really want is for it to be `"localhost"` (or whatever i specify via `-hostname`).  it was working that way before this change.

* any ideas on how to make it work again?
* thoughts on how to add a test to prevent a regression?
* ...and why do you treat `"::"` differently to `"0.0.0.0"`?  shouldn't they either both be changed or not changed.